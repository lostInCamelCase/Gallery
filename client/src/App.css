.container {
  /* display: grid; */
  grid-template-columns: repeat(auto-fill, minmax(200px, 1fr));
  grid-template-rows: 30px 30px 100px 100px 100px 100px auto;
  margin-left: 10%;
  margin-right: 10%;
  height: 100%;
}

.pictureContainer {
  display: grid;
  grid-template-rows: repeat(20% 20%);


}