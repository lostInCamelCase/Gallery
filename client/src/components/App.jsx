import React from 'react';
import axios from 'axios';
import GalleryList from './GalleryList.jsx';
import PictureView from './PictureView.jsx';
import TitleView from './TitleView.jsx';
import styles from '../App.css';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      pictures: [],
      gallery: this.props.infoState.gallery,
      ratings: [],
      stayName: '',
      show: false,
      currentIdx: 0,

    };
    this.hideModal = this.hideModal.bind(this);
    this.showModal = this.showModal.bind(this);
    this.leftArrow = this.leftArrow.bind(this);
    this.rightArrow = this.rightArrow.bind(this);
  }

  componentDidMount() {
    this.getStayName();
    this.getRatings();
    this.getPictures();
  }

  getStayName() {
    axios.get('/stay')
      .then((response) => {
        this.setState({ stayName: response.data[0] });
      })
      .catch((err) => {
        console.log('error', err);
      });
  }

  getPictures() {
    axios.get('/stay/pictures')
      .then((response) => {
        this.setState({ pictures: response.data }, () => {});
        this.setState({ gallery: response.data.slice(0, 6) }, () => {});
      })
      .catch((err) => {
        console.log('error', err);
      });
  }

  getRatings() {
    axios.get('/stay/ratings')
      .then((response) => {
        this.setState({ ratings: response.data[0] });
      })
      .catch((err) => {
        console.log('error', err);
      });
  }

  showModal() {
    this.setState({ show: true });
  }

  hideModal() {
    this.setState({ show: false });
  }

  leftArrow() {
    if (this.state.currentIdx > 0) {
      this.setState({ currentIdx: this.state.currentIdx - 1 });
    }
    if (this.state.currentIdx === 0) {
      this.setState({ currentIdx: 0 });
    }
  }

  rightArrow() {
    const length = this.state.pictures.length;
    if (this.state.currentIdx >= 0 && this.state.currentIdx < length - 1) {
      this.setState({ currentIdx: this.state.currentIdx + 1 });
    }
    if (this.state.currentIdx === length) {
      this.setState({ currentIdx: length });
    }
  }

  render() {
    return (
      <div className={styles.container}>
        <TitleView stayName={this.state.stayName} rating={this.state.ratings}/>
        <div className={styles.pictureContainer}>
        <PictureView gallery={this.state.gallery} openModal={this.showModal} />
        <GalleryList show={this.state.show} handleClose={this.hideModal} pictures={this.state.pictures} left={this.leftArrow} right={this.rightArrow} idx={this.state.currentIdx} />
        </div>
      </div>
    );
  }
}

export default App;
