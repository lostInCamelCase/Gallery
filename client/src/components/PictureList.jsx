import React from 'react';
import PictureView from './PictureView.jsx';

const PictureList = ({pictures}) => (
  <div>
   {/* {gallery.map((image) => (
    //  <PictureView image={image} key={image.id}/>
   ))} */}
  </div>
);

export default PictureList;
