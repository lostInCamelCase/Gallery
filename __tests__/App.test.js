import React from 'react';
import {mount, shallow} from 'enzyme';

import App from '../client/src/components/App.jsx'
import PictureList from '../client/src/components/PictureList.jsx'

describe('<App/>', () => {
  it('renders <PictureList/> component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })
})
