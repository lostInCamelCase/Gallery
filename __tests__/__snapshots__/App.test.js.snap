// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> component create snapshot of App component 1`] = `ShallowWrapper {}`;
